dependencies/aws-sdk-cpp/meson.build

```diff
@@ -0,0 +1,59 @@
+# aws s3 dependency file
+#============================================================================
+
+
+# Look for aws-cpp-sdk, an optional dependency
+#--------------------------------------------------
+aws_sdk_cpp_dep_list = [
+    dependency('aws-cpp-sdk-core', required: get_option('enable_s3')),
+    dependency('aws-cpp-sdk-s3', required: get_option('enable_s3')),
+    dependency('aws-cpp-sdk-transfer', required: get_option('enable_s3'))
+]
+
+aws_sdk_cpp_dep = declare_dependency(dependencies : aws_sdk_cpp_dep_list)
+
+# check if dependencies are found
+#--------------------------------------------------
+check_aws_sdk_cpp = true
```

Comment:
This is a bit more confusing than `has_`.
`has_`, `is_`, is a pretty common naming for booleans, because the name of the variable makes its value more explicit. On the other hand, "check" is a verb, so this variable kinda reads like a function...

Reply:
the comment directly above it removes any ambiguity

Reply:
A lot of comments are actually code smells: they often indicate that the code is not explicit, and add another dimension to take care of for maintenance (code AND comments).
And also, following this logic, we would need to comment every place this variable is used...

Reply:
integer comments are 99% of the time only listed when the variable is first defined. for checking if a device has a dependency, you should use the .found() subcommand of a dependency object instead

Reply:
You missed my point:
Comments are very rarely a good thing in code; Relying on comments to make explicit what a variable contains is a smell (because in practice: code / comments evolve differently, a confusing variable will confuse readers everytime its used - because only commented at declaration time).
Good naming will always trump commenting to remove ambiguity.

Reply:
changed now :)